src/Container/PackageProxyContainer.php

```diff
@@ -55,32 +44,32 @@ public function has(string $id): bool
     /**
      * @return bool
      *
-     * @throws \Exception
      * @psalm-assert-if-true ContainerInterface $this->container
+     * @psalm-assert-if-false null $this->container
      */
     private function tryContainer(): bool
     {
-        if ($this->container) {
+        if ($this->container !== null) {
             return true;
         }
 
         /** TODO: We need a better way to deal with status checking besides equality */
         if (
-            $this->package->statusIs(Package::STATUS_READY)
+            $this->package->statusIs(Package::STATUS_INITIALIZED)
+            || $this->package->statusIs(Package::STATUS_BOOTING)
```

Comment:
Why where those 2 added as conditions to compile the Container? Wondering if this will not break things when compiling the Container too early? The earliest point where you should be able to access the Container is in [`Package::doExecute()`](https://github.com/inpsyde/modularity/blob/master/src/Package.php#L534) which is after the [`Package::STATUS_MODULES_ADDED`](https://github.com/inpsyde/modularity/blob/master/src/Package.php#L407-L409) is set.

Reply:
This is a bug. We might fix it in another PR, but the bug still needs fixing. (Actually this kind of "escaped" my local code where I was experimenting with a few things).

Btw, since we introduced `build()` in v1.7.0, we can access the container after `build()`. A code like the following is totally fine:

```php
$package->build()->container();
```

After `build()` is called, the container can not be modified anymore so it is fine to get an instance of it. 

Before `build()` was there, you could get the container only after calling `boot()` so the code was fine because the Package status is `booted` by then.

However, we did not update this when we introduced `build()`, so now even if the connected package is built, so it would be very fine to get services from it, we throw an exception here.

Currently, this forces us to only call services from connected packages after they are booted, which might be very late. For this reason, we are kind of forced to boot packages early, even if not needed.

We introduced `build()` _exactly_ for this reason, but right now using `build()` early and `boot()` later is "dangerous" because if another package connects to your package they might get a fatal error.

Reply:
Okay, we fix/improve this in a separate PR upfront so that we can include here. 👍🏻This way we maybe also should update the `boot()` method and remove the param `Module ...$defaultModules` :)

Reply:
@Chrico I reverted this change (b7a7fe44) and will do another PR just for this.

Reply:
@Chrico here's the new PR that should be  merged before this :) https://github.com/inpsyde/modularity/pull/49